angular-hometask/src/app/core/breadcrumbs/breadcrumbs.component.ts

```diff
@@ -1,15 +1,62 @@
 import { Component, OnInit } from '@angular/core';
+import { Router, NavigationEnd, ActivatedRoute, PRIMARY_OUTLET, Params, UrlSegment } from '@angular/router';
+interface Breadcrumb {
+  label: string;
+  params: Params;
+  url: string;
+}
 
 @Component({
   selector: 'app-breadcrumbs',
   templateUrl: './breadcrumbs.component.html',
   styleUrls: ['./breadcrumbs.component.scss']
```

Comment:
For optimization purpose I advise u make changeDetectionRef as OnPush and then add ViewEncapsulation.None

Reply:
It did not work. I really have to refactor thoroughly the breadcrumbs component.